rdr_service/dao/participant_counts_over_time_service.py

```diff
@@ -146,6 +162,49 @@ def get_field_name(name):
 
             logging.info('Init temp table for metrics cron job.')
 
+    def build_participant_sql(self, params, columns):
+        p_table_path = f"{self.bq_project}.{self.participant_table}"
+        ps_table_path = f"{self.bq_project}.{self.participant_summary_table}"
+        job_config = bigquery.QueryJobConfig(query_parameters=params)
+        sql = self.QUERIES['participant_sql'].format(participant_bq_table=p_table_path,
+                                                     participant_summary_bq_table=ps_table_path,
+                                                     columns=columns)
+        return self.run_bq_query(sql, job_config)
+
+    def run_bq_query(self, sql: str, job_config=None):
+        # Run the job with exponential backoff to avoid 403: rateLimitExceeded Error
+        # https://cloud.google.com/bigquery/docs/troubleshoot-quotas,
+        max_tries = 10
+        for n in range(max_tries):
+            try:
+                query_job = self.client.query(sql, job_config)
+                return query_job
+            except Exception:  # pylint: disable=broad-except
+                if n == max_tries - 1:
+                    raise Exception
+                time.sleep(2 ** n)
+
+    def batch_insert_results(self, session, temp_table_name, results):
+        # Insert results into RDR table, implement batching of the results
+        records_to_insert, batch_size = [], 100
+
+        columns_str = ", ".join([a for a, b in self.TEMP_FIELDS])
+        values_str = ",".join([':' + a for a, b in self.TEMP_FIELDS])
+        batch_insert_stmt = ('insert into `{temp_table_name}` ({columns_str}) values ({values_str});'
```

Comment:
codeacy saying there is a possible sql injection here. may need to update this query?

Reply:
I basically just kept the existing possible SQL injection from line 110 in the old code. Was just going to leave the same format for the SQL queries in this job.